diff -rup seabios-1.7.0/Makefile mysea/Makefile
--- seabios-1.7.0/Makefile	2012-05-28 16:53:27.898842738 -0400
+++ mysea/Makefile	2012-05-28 16:50:37.307644939 -0400
@@ -221,7 +221,7 @@ $(OUT)vgabios.bin: $(OUT)vgabios.bin.raw
 
 src/%.hex: src/%.dsl ./tools/acpi_extract_preprocess.py ./tools/acpi_extract.py
 	@echo "Compiling DSDT"
-	$(Q)cpp -P $< > $(OUT)$*.dsl.i.orig
+	$(Q)cpp $(DSDT_CPP_FLAGS) -P $< > $(OUT)$*.dsl.i.orig
 	$(Q)$(PYTHON) ./tools/acpi_extract_preprocess.py $(OUT)$*.dsl.i.orig > $(OUT)$*.dsl.i
 	$(Q)iasl -l -tc -p $(OUT)$* $(OUT)$*.dsl.i
 	$(Q)$(PYTHON) ./tools/acpi_extract.py $(OUT)$*.lst > $(OUT)$*.off
diff -rup seabios-1.7.0/src/acpi-dsdt.dsl mysea/src/acpi-dsdt.dsl
--- seabios-1.7.0/src/acpi-dsdt.dsl	2012-04-14 22:27:40.000000000 -0400
+++ mysea/src/acpi-dsdt.dsl	2012-05-28 16:52:11.745575415 -0400
@@ -609,6 +609,7 @@ DefinitionBlock (
  * Suspend
  ****************************************************************/
 
+#ifdef DSDT_PM
     /*
      * S3 (suspend-to-ram), S4 (suspend-to-disk) and S5 (power-off) type codes:
      * must match piix4 emulation.
@@ -627,6 +628,7 @@ DefinitionBlock (
         Zero,  /* reserved */
         Zero   /* reserved */
     })
+#endif
     Name (\_S5, Package (0x04)
     {
         Zero,  /* PM1a_CNT.SLP_TYP */
